test/NuGet.Core.Tests/NuGet.Build.Tasks.Test/StaticGraphRestoreArgumentsTests.cs

```diff
@@ -0,0 +1,348 @@
+<ProjectConfiguration Project=""{392ea86f-3561-4083-b556-c9915a48cfa9}"" AbsolutePath=""D:\src\isomething\lib\Plugins\ReconCT\contoso.ReconCT.Plugins.WindmillFilter.UI\contoso.ReconCT.Plugins.WindmillFilter.UI.csproj"">Release|x64</ProjectConfiguration>
+<ProjectConfiguration Project=""{ab6258e0-abf1-4693-8af3-aaf8ee826450}"" AbsolutePath=""D:\src\isomething\Apps\DataTest\DataTestVisualizer\DataTestVisualizer.csproj"">Release|x64</ProjectConfiguration>
+<ProjectConfiguration Project=""{6c20dfa3-a9da-4db6-ad2f-22b2cf945909}"" AbsolutePath=""D:\src\isomething\lib\Plugins\ReconCT\contoso.ReconCT.Plugins.PNR\isomething.ReconCT.Plugins.PNR.csproj"">Release|x64</ProjectConfiguration>
+<ProjectConfiguration Project=""{810d6ed4-2a8f-4287-a314-100a834592f4}"" AbsolutePath=""D:\src\isomething\lib\Plugins\ReconCT\contoso.ReconCT.Plugins.PNR.UI\isomething.ReconCT.Plugins.PNR.UI.csproj"">Release|x64</ProjectConfiguration>
+<ProjectConfiguration Project=""{e7db0f20-d93c-4d61-9fc9-a9d292a2cb4c}"" AbsolutePath=""D:\src\isomething\lib\Plugins\ReconCT\isomething.ReconCT.Plugins.IRIS.Config\isomething.ReconCT.Plugins.IRIS.Config.csproj"">Release|x64</ProjectConfiguration>
+<ProjectConfiguration Project=""{4542180d-5a6c-47c2-815a-82eb776d028a}"" AbsolutePath=""D:\src\isomething\lib\Plugins\ReconCT\isomething.ReconCT.Plugins.PNR.Config\isomething.ReconCT.Plugins.PNR.Config.csproj"">Release|x64</ProjectConfiguration>
+<ProjectConfiguration Project=""{43c51672-39dc-4f4f-9009-42ac82b9f52b}"" AbsolutePath=""D:\src\isomething\lib\Plugins\ReconCT\isomething.ReconCT.Plugins.WindmillFilter.Config\isomething.ReconCT.Plugins.WindmillFilter.Config.csproj"">Release|x64</ProjectConfiguration>
+<ProjectConfiguration Project=""{269c7d4e-026e-43f6-825c-4161148c1e89}"" AbsolutePath=""D:\src\isomething\lib\Recon\isomething.Recon.Contracts\isomething.Recon.Contracts.csproj"">Release|x64</ProjectConfiguration>
+<ProjectConfiguration Project=""{eb1632dc-6286-42eb-a50a-46ad508b1c25}"" AbsolutePath=""D:\src\isomething\Apps\ReconCT\isomething.ReconCT.Xacb\isomething.ReconCT.Xacb.csproj"">Release|x64</ProjectConfiguration>
+<ProjectConfiguration Project=""{e86bf4e2-19e5-499d-9c58-e73bbf5cdf99}"" AbsolutePath=""D:\src\isomething\lib\Prep\isomething.Prep.Pipeline\isomething.Prep.Pipeline.csproj"">Release|x64</ProjectConfiguration>
+<ProjectConfiguration Project=""{f93f4640-8094-4f5f-a0db-d8a61ce014f4}"" AbsolutePath=""D:\src\isomething\lib\Prep\isomething.Prep.Algorithms.ScatterCorrection\isomething.Prep.Algorithms.ScatterCorrection.csproj"">Release|x64</ProjectConfiguration>
+<ProjectConfiguration Project=""{50448e09-f547-4c6a-9a30-addbc70f4b2d}"" AbsolutePath=""D:\src\isomething\lib\ImageProcessing\Registration\isomething.ImageProcessing.Algorithms.Registration.Scr.Test\isomething.ImageProcessing.Algorithms.Registration.Scr.Test.csproj"">Release|x64</ProjectConfiguration>
+</SolutionConfiguration>",
```

Comment:
This doesn't need 230 lines to validate whatever this test wants to validate. It does make reading the file/test painful, however. Just one or two projects in the test solution would be sufficient.

Reply:
So technically this all came about because there was a huge global property set.  My inclination was to have a test that verified that a large amount of global properties won't mess anything up.  But to your point, do I really need the test if I'm no longer sending the data as command-line arguments since the new implementation shouldn't be affected by limits?  

What does everyone think, is it worth testing that this whole thing can handle a large set of global properties?

Reply:
A few comments:

1. This test isn't running the code that actually passes the arguments to another process. Hence, if the test was written first (Test-Driven-Development), the test wouldn't show the old behaviour/implementation failing. Therefore, this test wouldn't catch a regression. What it's really testing is if serialization/deserialization can round-trip the data. You don't need a 32k char test to validate serialization round-tripping.

I've been a big fan of trying to get as close to 100% code coverage for a long time, but recently I've been thinking about how easy/difficult it is to refactor internal code, so lately I've been thinking more about "will this test actually catch a regression" and "does this test validate API contracts/customer experience, or is it validating the current implementation"


2. If you feel passionate about testing a 32k char string, you can move it into a separate method, so the method with the `[Test]`/`[Fact]`/`[Theory]` attribute remains short and easier to read. The name of the method, or comments in the test, can then explain why testing 32k characters is important.  Despite having the context of this PR/bug, it wasn't clear to me why this test was using such long data. It must be harder for others who will read it after it's merged to understand.

3. The XML could be generated in a loop, something like:

```cs
int projectNumber = 0;
while (stringBuilder.Length < 33000)
{
  string newConfiguration = $"<ProjectConfiguration Project=\"{Guid.NewGuid()}\" AbsolutePath=\"d:\src\something\project{projectNumber}\project{projectNumber}.csproj\">Release|x64</ProjectConfiguration>";
  projectNumber++;
  stringBuilder.AppendLine(newConfiguration);
}
```

That's much fewer lines than the test data. Though I'd argue that it still needs a comment as to why it's generating a string longer than 33k, since I don't think it's intuitive why a > 32k string is needed.

Reply:
I'll just reduce the test data, originally I wanted to make sure my JSON or text parsing logic could handle complex characters but now that I'm using binary serialization I don't think its useful.